File: core/src/main/java/com/xmartlabs/base/core/common/EntityProvider.java
package com.xmartlabs.base.core.common;

import android.support.annotation.Nullable;

import com.annimon.stream.Optional;

/** Used to provide an entity. */
public interface EntityProvider<T> {
  /**
   * Provides the entity.
   *
   * @return the entity
   */
  Optional<T> provideEntity();

  /**
   * Updates the entity in the database.
   *
   * @param entity Entity to be saved
   */
  void updateEntity(@Nullable T entity);

  /** Deletes the entity in the database. */
  void deleteEntity();
}

File: core/src/main/java/com/xmartlabs/bigbang/core/common/EntityProvider.kt
package com.xmartlabs.bigbang.core.common

/** Used to provide an entity.  */
interface EntityProvider<T> {
  /**
   * Provides the entity.
   * @return the entity
   */
  fun provideEntity(): T?

  /**
   * Updates the entity in the database.
   * @param entity Entity to be saved
   */
  fun updateEntity(entity: T)

  /** Deletes the entity in the database.  */
  fun deleteEntity()
}
